helloworld/client.go

```diff
@@ -17,54 +17,51 @@ package helloworld
 
 import (
 	"context"
+	"fmt"
 
 	trillian "github.com/google/trillian"
 	"github.com/google/trillian/merkle/logverifier"
 	"github.com/google/trillian/merkle/rfc6962/hasher"
 
 	"github.com/google/trillian-examples/formats/log"
+	"golang.org/x/mod/sumdb/note"
 )
 
 // Personality is the front-end for the Trillian log.
 type Personality interface {
 	// Append stores an entry in the log and, once it is there, returns a
 	// new checkpoint that commits to the new entry (in addition to all
 	// previous ones).
-	Append(context.Context, []byte) (*log.Checkpoint, error)
+	Append(context.Context, []byte) ([]byte, error)
 
-	// ProveIncl proves the inclusion of a given entry.  It is the job of
-	// the client to verify this inclusion proof in order to convince
+	// ProveIncl proves the inclusion of a given entry in a given checkpoint size.
+	// It is the job of the client to verify this inclusion proof in order to convince
 	// itself that the entry really has been included in the log.
-	ProveIncl(context.Context, *log.Checkpoint, []byte) (*trillian.Proof, error)
+	ProveIncl(context.Context, uint64, []byte) (*trillian.Proof, error)
 
 	// UpdateChkpt provides a new checkpoint and proves it is consistent
 	// with an old one.  Again, it is the job of the client to verify this
 	// consistency proof.
-	UpdateChkpt(context.Context, *log.Checkpoint) (*log.Checkpoint, *trillian.Proof, error)
+	UpdateChkpt(context.Context, uint64) ([]byte, *trillian.Proof, error)
```

Comment:
The comment doesn't match the signature.

Reply:
Not sure what specific change you're asking for here - it still returns a checkpoint (albeit flattened), and a proof for an old checkpoint? But I suspect that part of the problem here is that the params on the interface methods don't have names, I'll fix that.

Reply:
It seems different this afternoon, but this morning "proves it is consistent with an old one" didn't seem appropriate when what was passed in was a `uint64`. Having names here helps.